pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -1390,6 +1407,9 @@ private void sendQuery(Query query, V3ParameterList parameters, int maxRows, int
 
       // If we saw errors, don't send anything more.
       if (resultHandler.getException() == null) {
+        if (fetchSize != 0) {
```

Comment:
This seems like it could use some explanation as to why we are adding this query on error

Reply:
As I understand the code, in this situation we are adding this query when there is no error (that's why result, from getException method, is checked to be null).